File: server/API/Models/Board.cs
namespace VoterAPI.Models;

public class Board
{
    public int Id { get; set; }
    public required string Title { get; set; }
    public required string Description { get; set; }
    public DateTime CreatedDate { get; set; } = DateTime.UtcNow;
    public int CreatedByUserId { get; set; }
    public bool IsSuggestionsOpen { get; set; } = true;
    public bool IsVotingOpen { get; set; } = true;
    public bool RequireApproval { get; set; } = false;
    public VotingType VotingType { get; set; } = VotingType.Single;
    public int? MaxVotes { get; set; }
    public bool IsClosed { get; set; } = false;

    // Navigation properties
    public User? CreatedBy { get; set; }
    public ICollection<Suggestion> Suggestions { get; set; } = new List<Suggestion>();
}

File: client/src/types/board.ts
import { Suggestion } from './suggestion';

export interface Board {
  id: number;
  title: string;
  description: string;
  createdDate: string;
  createdByUserId: number;
  createdByUsername?: string;
  isSuggestionsOpen: boolean;
  isVotingOpen: boolean;
  requireApproval: boolean;
  votingType: 'Single' | 'Multiple';
  maxVotes?: number;
  isClosed: boolean;
  suggestionCount: number;
  totalVotes: number;
}

export interface BoardDetail extends Omit<Board, 'suggestionCount' | 'totalVotes'> {
  suggestions: Suggestion[];
}

export interface CreateBoardRequest {
  title: string;
  description: string;
  isSuggestionsOpen?: boolean;
  isVotingOpen?: boolean;
  requireApproval?: boolean;
  votingType: 'Single' | 'Multiple';
  maxVotes?: number;
}

export interface UpdateBoardRequest {
  title: string;
  description: string;
  isSuggestionsOpen: boolean;
  isVotingOpen: boolean;
  requireApproval: boolean;
  votingType: 'Single' | 'Multiple';
  maxVotes?: number;
}
